pkg/services/ngalert/state/manager.go

```diff
@@ -400,3 +422,12 @@ func (st *Manager) deleteStaleStatesFromCache(ctx context.Context, logger log.Lo
 func stateIsStale(evaluatedAt time.Time, lastEval time.Time, intervalSeconds int64) bool {
 	return !lastEval.Add(2 * time.Duration(intervalSeconds) * time.Second).After(evaluatedAt)
 }
+
+func tryToTakeAnImage(ctx context.Context, l log.Logger, c *ImageCapturer, r *ngModels.AlertRule, resolve func(img *ngModels.Image)) {
```

Comment:
Although I can see a value in this refactoring, I am not sure this PR is a good place for it. Let's keep this PR focused on one thing and do as less refactroings as possible. Please create a separate PR and we can look into it more closer.